components/tikv_util/src/quota_limiter.rs

```diff
@@ -0,0 +1,225 @@
+// Copyright 2022 TiKV Project Authors. Licensed under Apache-2.0.
+
+use std::time::Duration;
+
+use super::config::ReadableSize;
+use super::time::Limiter;
+use super::timer::GLOBAL_TIMER_HANDLE;
+
+use cpu_time::ThreadTime;
+use futures::compat::Future01CompatExt;
+
+// The cpu time is not a real statistics, only part of the processing logic is
+// taken into account, so it needs to be multiplied by a factor.
+// transfer milli cpu to micro cpu
+//
+// TODO: Don't adjusted based on experience.
+const CPU_TIME_FACTOR: f64 = 0.8;
+
+// To avoid long tail latency.
+const MAX_QUOTA_DELAY: Duration = Duration::from_secs(1);
+
+// Quota limiter allows users to obtain stable performance by increasing the
+// completion time of tasks through restrictions of different metrics.
+#[derive(Debug)]
+pub struct QuotaLimiter {
+    cputime_limiter: Limiter,
+    write_bandwidth_limiter: Limiter,
+    read_bandwidth_limiter: Limiter,
+}
+
+// Throttle must be consumed in quota limiter.
+pub struct Throttle {
```

Comment:
Considering the context,
```suggestion
pub struct Sample {
```